### PR TITLE
Add OpenAPI 3 with Docker

### PR DESCRIPTION
**Pull request by hoangmirs**
## What happened 👀

- Add OpenAPI implementation.
- Add login API documentation.

## Insight 📝

- We set up OpenAPI with docker (`swaggerapi/swagger-ui`) also structure for the openapi folder.
- To use the Swagger UI, please go to: `http://localhost:8080/`. After deploying, we need to use the port `8080` to access that page.

## Proof Of Work 📹

Screenshot 2023-06-08 at 11 11 37

### CODE REVIEW

File: docs/resources/oauth/token.yml [hunk 1]
```diff
@@ -0,0 +1,79 @@
+summary: |-
+  Request access token
+
+description: |-
+  Request an Access Token using either a client-side obtained OAuth 2.0
+  authorization code or a server-side JWT assertion.
+
+  An Access Token is a string that enables Box to verify that a
```
Comment (olivierobert) [inline docs/resources/oauth/token.yml:8]: `Box` what does it refer to? Is that from a generator 😅 ?

Comment (hoangmirs) [reply docs/resources/oauth/token.yml:8]: 🙈 , I'm refer the [Box-OpenAPI](https://github.com/box/box-openapi/blob/main/content/openapi.yml), so I copied that text from their docs 😭

Comment (suho) [reply docs/resources/oauth/token.yml]: @olivierobert @hoangmirs I updated the docs in https://github.com/nimblehq/nimble-survey-web/pull/86/commits/41f9d6756709dadf7514804f5741ce1f2c34ca92

**Comment by suho** [comment]
@olivierobert We are going to add all API documents with OpenAPI, so I think we can merge OpenAPI codes to the `develop` branch.

**Review by suho — APPROVED** [review]

**Pull request merged**